src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3949,19 +3950,23 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
                 MemberResolutionResult<MethodSymbol> memberResolutionResult;
                 ImmutableArray<MethodSymbol> candidateConstructors;
-                if (TryPerformConstructorOverloadResolution(
-                    initializerType,
-                    analyzedArguments,
-                    WellKnownMemberNames.InstanceConstructorName,
-                    errorLocation,
-                    false, // Don't suppress result diagnostics
-                    diagnostics,
-                    out memberResolutionResult,
-                    out candidateConstructors,
-                    allowProtectedConstructorsOfBaseType: true))
+                bool found = TryPerformConstructorOverloadResolution(
+                                    initializerType,
+                                    analyzedArguments,
+                                    WellKnownMemberNames.InstanceConstructorName,
+                                    errorLocation,
+                                    false, // Don't suppress result diagnostics
+                                    diagnostics,
+                                    out memberResolutionResult,
+                                    out candidateConstructors,
+                                    allowProtectedConstructorsOfBaseType: true);
+                MethodSymbol resultMember = memberResolutionResult.Member;
+
+                validateRecordCopyConstructor(containingType, constructor, baseType, resultMember, errorLocation, diagnostics);
```

Comment:
>validateRecordCopyConstructor(containingType, constructor, baseType, resultMember, errorLocation, diagnostics); [](start = 16, length = 111)

I don't think we should be doing this if overload resolution failed, it already complained in that case. We should verify only success case.

Reply:
`CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_NoPositionalMembers` illustrates why it is useful to check when overload resolution fails.
Diagnostic #1 alone is not user-friendly (does not hint what is actually the problem or how to fix it). Diagnostic #2 serves that purpose.

```
        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor()
        {
            var source =
@"public record B(object N1, object N2)
{
}
public record C(object P1, object P2) : B(0, 1)
{
    public C(C c) // 1, 2
    {
    }
}
";
            var comp = CreateCompilation(source);
            comp.VerifyDiagnostics(
                // (6,12): error CS1729: 'B' does not contain a constructor that takes 0 arguments
                //     public C(C c) // 1, 2
                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "C").WithArguments("B", "0").WithLocation(6, 12),
                // (6,12): error CS8868: A copy constructor in a record deriving from another record must invoke that base record's copy constructor 'B.B(B)'.
                //     public C(C c) // 1, 2
                Diagnostic(ErrorCode.ERR_CopyConstructorMustInvokeBaseCopyConstructor, "C").WithArguments("B.B(B)").WithLocation(6, 12)
                );
        }

```

---
In reply to: [439016789](https://github.com/dotnet/roslyn/pull/45022#discussion_r439016789) [](ancestors = 439016789)

Reply:
In my opinion it doesn't serve the purpose and the first error is sufficient and to the point. But I can live with this.

---
In reply to: [439077473](https://github.com/dotnet/roslyn/pull/45022#discussion_r439077473) [](ancestors = 439077473,439016789)